Sources/protoc-gen-swift/MessageLayoutCalculator.swift

```diff
@@ -37,10 +37,13 @@ struct MessageLayoutCalculator {
     ///
     /// The first element in this array corresponds to the submessage with index 1, and the rest
     /// increase accordingly.
-    var submessageNames: [String] {
-        submessages.usedSubmessages.sorted { $0.value < $1.value }.map { $0.key }
+    var submessages: [SubmessageInfo] {
+        submessageCollector.usedSubmessages.sorted { $0.value.index < $1.value.index }.map { $0.value }
     }
 
+    /// Indicates whether we need to generate an `isInitialized` property for this message.
+    private(set) var needsIsInitializedGeneration = false
```

Comment:
I guess the one thing missing is feeding in if there are extension ranges.  While the layout code is a TODO, we also need to factor it into if we need to generate the override.

Reply:
I deleted this since it was only used by `MessageGenerator` and that was replaced with just `containsRequiredFields`.